packages/rocketchat-lib/server/functions/sendMessage.js

```diff
@@ -1,10 +1,69 @@
 import _ from 'underscore';
 
+const validateBodyAttachments = (attachments) => {
+
+	const validateAttachmentsFields = (attachmentFields) => {
+		check(attachmentFields, Match.ObjectIncluding({
+			short: Match.Maybe(Boolean),
+			title: String,
+			value: String
+		}));
+	};
+
+	const validateAttachment = (attachment) => {
+		check(attachment, Match.ObjectIncluding({
+			color: Match.Maybe(String),
+			text: Match.Maybe(String),
+			ts: Match.Maybe(String),
+			thumb_url: Match.Maybe(String),
+			message_link: Match.Maybe(String),
+			collapsed: Match.Maybe(Boolean),
+			author_name: Match.Maybe(String),
+			author_link: Match.Maybe(String),
+			author_icon: Match.Maybe(String),
+			title: Match.Maybe(String),
+			title_link: Match.Maybe(String),
+			title_link_download: Match.Maybe(Boolean),
+			image_url: Match.Maybe(String),
+			audio_url: Match.Maybe(String),
+			video_url: Match.Maybe(String)
+		}));
+
+		if (attachment.fields.length) {
+			attachment.fields.map(validateAttachmentsFields);
+		}
+	};
+
+	attachments.map(validateAttachment);
+};
+
 RocketChat.sendMessage = function(user, message, room, upsert = false) {
 	if (!user || !message || !room._id) {
 		return false;
 	}
 
+	try {
+		check(message, Match.ObjectIncluding({
+			msg: Match.Maybe(String),
+			text: Match.Maybe(String),
+			alias: Match.Maybe(String),
+			emoji: Match.Maybe(String),
+			avatar: Match.Maybe(String),
+			attachments: Match.Maybe(Array)
+		}));
+
+		if (Array.isArray(message.attachments) && message.attachments.length) {
+			validateBodyAttachments(message.attachments);
+		}
+	} catch (error) {
```

Comment:
Why are we catching an error here to then throw the same error? Let's not do that. Either allow the error to be thrown or throw a customized error.